### PR TITLE
Issue/1.3.1 2.11

### PR DESCRIPTION
**Pull request by skyluc**
Update to Scala 2.11.6 to fix the REPL tests.
Fix to compile on 2.116.
Tweaks to mima config, for Scala 2.11.

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://ci.typesafe.com/job/ghprb-spark/19/

**Build Log**
_last 10 lines_

```
[...truncated 8023 lines...]
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    at java.lang.Thread.run(Thread.java:745)
[error] (streaming-kafka/*:mimaPreviousClassfiles) sbt.ResolveException: unresolved dependency: org.apache.spark#spark-streaming-kafka_2.11;1.3.0: not found
[error] Total time: 52 s, completed May 6, 2015 2:52:54 PM
[error] Got a return code of 1 on line 222 of the run-tests script.
Build step 'Execute shell' marked build as failure
Notifying upstream projects of job completion
Setting status of efca8625ff3846e4d9c4598b7b2ed42e2408f9b0 to FAILURE with url http://ci.typesafe.com/job/ghprb-spark/19/ and message: Merged build finished.
```

Test FAILed.

File: project/MimaBuild.scala [hunk 1]
```diff
@@ -91,8 +91,8 @@ object MimaBuild {
 
   def mimaSettings(sparkHome: File, projectRef: ProjectRef) = {
     val organization = "org.apache.spark"
-    val previousSparkVersion = "1.2.0"
-    val fullId = "spark-" + projectRef.project + "_2.10"
+    val previousSparkVersion = "1.3.0"
+    val fullId = "spark-" + projectRef.project + "_2.11"
```
Comment (dragos) [inline project/MimaBuild.scala:95]: Shouldn't this be controlled by the execution flags (or `change-version-to-2.11.sh`)? Note that the validator runs the `2.10` build but mima seems to look at 2.11.

Comment (skyluc) [inline project/MimaBuild.scala:95]: We should be able to use sbt Scala version support for that.

Comment (dragos) [inline project/MimaBuild.scala:95]: What do you mean? :)

**Pull request closed without merging**